EasyLearn/app/controllers/videos_controller.rb

```diff
@@ -0,0 +1,46 @@
+require "faraday"
+
+class VideosController < ApplicationController
+  before_action :require_login
+
+  def index
+  end
+  
+  def new
+    @video = Video.new
+    @categories = Category.all
+  end
+  
+  def create
+    @videoID = params[:video][:link]
+
+    if @videoID.empty?
+      flash[:alert] = "The fields cannot be empty"
+      redirect_to new_video_path
+    else
+      @response = Faraday.get("https://youtube.googleapis.com/youtube/v3/videos?part=snippet&id=#{@videoID}&key=AIzaSyC-039WYenwbFNwXQRAd3ynWW9AeEVn1rs")
```

Comment:
Is the key you have in this line a private key? If so, I suggest you put it in an environment variable. Remember that private keys should not appear on GitHub.

Reply:
I aggre with @YilKanda

Reply:
Done, already I change the key to an environment variable usign figaro gem.